### PR TITLE
Add ability to manage image upload via Cloudinary #to-upstream

### PR DESCRIPTION
**Pull request by Stvad**
This adds an ability to use Cloudinary as an image upload target for images inserted in the editor. 
Mostly migrating the code from WakingUP forum.

If the CKEditor image upload url setting is specified (as it is for EA forum and LW) - that will take a priority and images will be uploaded to CKEditor.

Otherwise code falls back on using Cloudinary.
If neither is configured - user will see an alert saying that Cloudinary setting needs to be configured (but editor will keep working otherwise)

### CODE REVIEW

File: packages/lesswrong/components/editor/CKCommentEditor.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { ckEditorUploadUrlSetting, ckEditorWebsocketUrlSetting } from '../../lib
 import { ckEditorUploadUrlOverrideSetting, ckEditorWebsocketUrlOverrideSetting, forumTypeSetting } from '../../lib/instanceSettings';
 import { defaultEditorPlaceholder } from '../../lib/editor/make_editable';
 import { mentionPluginConfiguration } from "../../lib/editor/mentionsConfig";
+import {cloudinaryConfig} from '../../lib/editor/cloudinaryConfig.ts'
```
Comment (b0b3rt) [inline packages/lesswrong/components/editor/CKCommentEditor.tsx:10]: I am surprised that whatever our module & build configuration is, it allows importing both with and without file extensions... but for now I think I'd prefer to keep it consistent (without)

Comment (Stvad) [reply packages/lesswrong/components/editor/CKCommentEditor.tsx:10]: will update, annoyingly my IDE has a bug that all auto-imports have an extension regardless of configuration =\

**Review by Will-Howard — COMMENTED** [review]
I'm probably just missing something but I can't get it to use cloudinary over the default ckeditor cdn. I've confirmed that `CloudinaryAdapterPlugin` is called, but `upload()` isn't being called when I upload an image via the toolbar

**Comment by Stvad** [comment]
@Will-Howard Cloudinary will be used if the CKEditor CDN is not configured (and appropriate Cloudinary settings are), there is currently no facility to force it to use Cloudinary when CKEditor CDN is configured, as the main use-case for this is to enable image upload in the editor when you are not paying for CKEditor cloud.

**Review by Will-Howard — APPROVED** [review]

**Pull request merged**